### ISSUE TITLE
Error Inquiry - PlanBuild or MultiUserChest

### ISSUE THREAD

**Issue by ValhallaCommunity** [MSchmoecker/No-Chest-Block#7]
Hey MSchmoecker, I can't tell if this error is related to PlanBuild or MultiUserChest. 

Both mods still work.

```
[Error  : Unity Log] ArgumentException: An item with the same key has already been added. Key: -1394152847
Stack trace:
System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) (at <695d1cc93cca45069c528c15c9fdd749>:0)
ZNetView.Register[T] (System.String name, System.Action`2[T1,T2] f) (at <afd4e3f8ab8b460f8ff25d476fbd573e>:0)
MultiUserChest.Patches.ContainerPatch.RegisterRPCs (Container __instance) (at D:/Projects/No-Chest-Block/MultiUserChest/Patches/ContainerPatch.cs:40)
MultiUserChest.Patches.ContainerPatch.ContainerAwakePatch (Container __instance) (at D:/Projects/No-Chest-Block/MultiUserChest/Patches/ContainerPatch.cs:33)
(wrapper dynamic-method) Container.DMD<Container::Awake>(Container)
UnityEngine.Object:Instantiate(GameObject)
Player:DMD<Player::SetupPlacementGhost>(Player)
MonoMod.Utils.DynamicMethodDefinition:Trampoline<Player::SetupPlacementGhost>?-977103872(Player)
PlanBuild.Blueprints.BlueprintManager:Player_SetupPlacementGhost(orig_SetupPlacementGhost, Player)
MonoMod.Utils.DynamicMethodDefinition:Hook<Player::SetupPlacementGhost>?-317712000(Player)
MonoMod.Utils.DynamicMethodDefinition:Trampoline<Player::SetupPlacementGhost>?993413120(Player)
PlanBuild.Plans.PlanManager:Player_SetupPlacementGhost(orig_SetupPlacementGhost, Player)
MonoMod.Utils.DynamicMethodDefinition:Hook<Player::SetupPlacementGhost>?687405696(Player)
Player:SetSelectedPiece(Vector2Int)
Hud:SelectPiece(UIInputHandler)
Hud:OnLeftClickPiece(UIInputHandler)
UIInputHandler:OnPointerDown(PointerEventData)
UnityEngine.EventSystems.EventSystem:Update()
```

Thanks for taking a peak!

**Comment by MSchmoecker** [comment]
Do you have the OdinShipPlus mod installed? This error is because of an conflict with it. OdinShipPlus creates two containers on ships which is not done in vanilla and instead uses custom code. I'm working on a fix at the moment to exclude those ships from the interaction with MUC.
If you don't have the mod installed, what piece do you select when the error pops up? At least the stacktrace indicates that this is thrown when doing this.

**Comment by ValhallaCommunity** [comment]
Hey MSchmoecker, we do have OdinShipPlus mod install on the server. Is this something that could be a better player experience if we coordinated with Marlthon (mod author of OSP)? Happy to bridge the connection, just not familiar with the error and solution for that best experience.

Thanks for taking time to consider and look at the error!

**Comment by MSchmoecker** [comment]
I've already spoke with Marlthon an we concluded that this has to be fixed on the MUC side. The solution will be to exclude the OSP ships from interacting with MUC.

**Comment by MSchmoecker** [comment]
Fixed in 0.4.4. Ships with multiple containers from OdinShipPlus are excluded from MUC and behave as they normally would in the mod.
